indico/legacy/pdfinterface/conference.py

```diff
@@ -580,7 +580,12 @@ def _processDayEntries(self, day, story):
                 ts = deepcopy(originalts)
                 contribs = sorted(sess_block.contributions, key=attrgetter('timetable_entry.start_dt'))
                 if sess_block.session.is_poster:
-                    for contrib in contribs:
+                    def sort_key_func(x):
+                        try:
+                            return int(x.board_number)
+                        except ValueError:
+                            return x
+                    for contrib in sorted(contribs, key=sort_key_func):
```

Comment:
Use `key=natural_sort_key` (and import that function); it'll automatically do natural sorting regardless of the text, so it will work nicely for both numbers, strings and strings containing numbers.